tree/30_generic_methods/condition_from_command.cf

```diff
@@ -0,0 +1,59 @@
+#####################################################################################
+# Copyright 2013 Normation SAS
+#####################################################################################
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, Version 3.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+#####################################################################################
+
+# @name Condition from command 
+# @description Execute a command and create outcome classes depending on its exit code
+#
+# @parameter condition_prefix   The condition name, use ${class_prefix} to create a local condition
+# @parameter command            The command to run
+# @parameter true_codes         List of codes that produce a true status separated with commas (ex: 1,2,5)
+# @parameter false_codes        List of codes that produce a false status separated with commas (ex: 3,4,6)
+# 
+# @class_prefix condition_from_command
+# @class_parameter condition_name
+#
+# This bundle will define a class condition_from_command_${command_name}_{kept,not_ok,ok,reached}
+# depending on the exit codes given in parameters.
+#
+# If the exit code is in the list, this will always produce a kept outcome class.
+# If the exit code is not in the list, this will produce an error outcome class.
+# If the exit code is in the list, this will additionnaly produce a ${condition_prefix}_true or a ${condition_prefix}_false condition.
+#
+# The created condition (class in cfengine speaking) is global to the agent.
+# To make the condition specific to current technique instance (or directive), use ${class_prefix} instead of a static name for the condition prefix
```

Comment:
This last sentence could be improve like
To make the condition specific to the current technique instance (or directives), add ${class_prefix} within the condition_prefix.